core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java

```diff
@@ -126,7 +126,9 @@ private HttpResponse convertHeaders(ResponseHeaders headers, boolean endStream,
         } else {
             res = new DefaultHttpResponse(HttpVersion.HTTP_1_1, nettyStatus, false);
             convertHeaders(headers, res.headers(), isTrailersEmpty);
-            maybeSetTransferEncoding(res);
+            if (!HttpStatus.isContentAlwaysEmpty(statusCode)) {
+                maybeSetTransferEncoding(res);
+            }
```

Comment:
Should we instead modify `maybeSetTransferEncoding()` to require a status code as an additional parameter?

Reply:
Thanks! updated. 😉